protocol/network/2-state-of-the-art.md

```diff
@@ -0,0 +1,70 @@
+2 An analysis the State of the Art in Network Stacks
+====================================================
+
+This section presents to the reader an analysis of the available protocols and arquitectures for a Network Stack. The goal is to provide the foundations from which to infer the conclusions and understand what are libp2p requirements and its designed arquitecture.
+
+## 2.1 The client-server model
+
+The client-server model indicates that both parties that ends of the channel have different roles, that support different services and/or have different capabilities, or in another words, speak different protocols.
+
+Building client-server applications has been natural tendency for a number of reasons:
+
+- The bandwidth inside a DataCenter is considerably high compared to the one available for clients to connect between each other
+- DataCenter resources are considerably cheaper, due to efficient usage and bulk stocking
+- Enables easier methods for the developer and system admin to have a fine grained control over the application
+- Reduces the number of heteregeneus systems to be handled (although it is still considerably high)
+- Systems like NAT make it really hard for client machines to find and talk with each other, forcing a developer to perform very clever hacks to traverse these obstacles.
+- Protocols started to be designed with the assumption that a developer will create a client-server application from the start.
+
+We even learned how to hide all of the complexity of a distributed system behind gateways on the Internet, using protocols that were designed to perform a point-to-point operation, such as HTTP, making it opaque for the application to see and understand how the cascade of service calls made for each request.
+
+`libp2p` offers a move towards dialer-listener interactions, from the client-server listener, where it is not implicit which of the entities, dialer or listener, has which capabilities or is enabled to perform which actions. Setting up a connection between two applications today is a multilayered problem to solve, and these connections should not have a purpose bias, instead support to several other protocols to work on top of the established connection. In a client-server model, a server sending data without a prior request from the client is known as a push model, which typically adds more complexity, in a dialer-listener model, both entities can perform requests independently.
+
+## 2.2 Categorizing the network stack protocols by solutions
```

Comment:
note: Same here. I've added, this comment didn't get 'outdated' because it was a couple of lines below